pkg/refactor/logengine/logger.go

```diff
@@ -0,0 +1,118 @@
+package logengine
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/direktiv/direktiv/pkg/flow/database/recipient"
```

Comment:
Why this package depends on something from /pkg/flow/database/ ?

Reply:
this is temporary, it will change when i will integrate the new logger